app/models/session.js

```diff
@@ -6,132 +6,126 @@ import CategorizableModel from 'ilios/mixins/categorizable-model';
 
 const { computed, isEmpty, isPresent, RSVP } = Ember;
 const { alias, mapBy, notEmpty, sum } = computed;
-const { PromiseArray, PromiseObject } = DS;
-
-var Session = DS.Model.extend(PublishableModel, CategorizableModel, {
-  title: DS.attr('string'),
-  attireRequired: DS.attr('boolean'),
-  equipmentRequired: DS.attr('boolean'),
-  supplemental: DS.attr('boolean'),
-  attendanceRequired: DS.attr('boolean'),
-  updatedAt: DS.attr('date'),
-  sessionType: DS.belongsTo('session-type', {async: true}),
-  course: DS.belongsTo('course', {async: true}),
-  ilmSession: DS.belongsTo('ilm-session', {async: true}),
-  objectives: DS.hasMany('objective', {async: true}),
-  meshDescriptors: DS.hasMany('mesh-descriptor', {async: true}),
-  sessionDescription: DS.belongsTo('session-description', {async: true}),
-  learningMaterials: DS.hasMany('session-learning-material', {async: true}),
-  offerings: DS.hasMany('offering', {async: true}),
-  administrators: DS.hasMany('user', {
+const { attr, belongsTo, hasMany, Model } = DS;
+const { all, Promise } = RSVP;
+
+export default Model.extend(PublishableModel, CategorizableModel, {
+  title: attr('string'),
+  attireRequired: attr('boolean'),
+  equipmentRequired: attr('boolean'),
+  supplemental: attr('boolean'),
+  attendanceRequired: attr('boolean'),
+  updatedAt: attr('date'),
+  sessionType: belongsTo('session-type', {async: true}),
+  course: belongsTo('course', {async: true}),
+  ilmSession: belongsTo('ilm-session', {async: true}),
+  objectives: hasMany('objective', {async: true}),
+  meshDescriptors: hasMany('mesh-descriptor', {async: true}),
+  sessionDescription: belongsTo('session-description', {async: true}),
+  learningMaterials: hasMany('session-learning-material', {async: true}),
+  offerings: hasMany('offering', {async: true}),
+  administrators: hasMany('user', {
     async: true,
     inverse: 'administeredSessions'
   }),
   isIndependentLearning: notEmpty('ilmSession.content'),
   offeringLearnerGroups: mapBy('offerings', 'learnerGroups'),
   offeringLearnerGroupsLength: mapBy('offeringLearnerGroups', 'length'),
   learnerGroupCount: sum('offeringLearnerGroupsLength'),
-  sortedOfferingsByDate: computed('offerings.@each.startDate', {
-    get() {
-      let defer = RSVP.defer();
+
+  /**
+   * All offerings for this session, sorted by offering startdate in ascending order.
+   * @property sortedOfferingsByDate
+   * @type {Ember.computed}
+   */
+  sortedOfferingsByDate: computed('offerings.@each.startDate', function() {
+    return new Promise(resolve => {
       this.get('offerings').then(offerings => {
-        let sortedOfferingsByDate = offerings.filter(offering => isPresent(offering.get('startDate'))).sort((a, b) => {
+        resolve(offerings.filter(offering => isPresent(offering.get('startDate'))).sort((a, b) => {
```

Comment:
I think wrapping this in a resolve gets a little bit hard to read.  Might be easier to read if it assigned to var, filtered and assigned to var, then sorted and assigned, then resolved.  Is that too verbose?